core/api/src/modules/ops/event.ts

```diff
@@ -139,4 +136,22 @@ export namespace EventOps {
       }
     }
   }
+
+  async function createProfileFromAnonymousId(
```

Comment:
This looks like what we had on the call, but it wasn't enough to prevent the deadlocks... was there something else?

Reply:
it worked with this and the `catch` left in. this made sure it was in both spots.